netty/src/main/java/io/grpc/netty/NettyServerTransport.java

```diff
@@ -164,12 +170,35 @@ private void notifyTerminated(Throwable t) {
     }
   }
 
+  @Override
+  @Nullable
+  public TransportTracer.Stats getTransportStats() {
+    if (channel.eventLoop().inEventLoop()) {
+      // This is necessary, otherwise we will block forever on the promise in the else case
+      return transportTracer.getStats();
+    }
+    final Promise<TransportTracer.Stats> promise = channel.eventLoop().newPromise();
+    channel.eventLoop().execute(new Runnable() {
+      @Override
+      public void run() {
+        promise.setSuccess(transportTracer.getStats());
+      }
+    });
+    try {
+      return promise.get();
```

Comment:
probably should set some timeout here, perhaps 1 minute or something just for safety

Reply:
Done